Nodejs/Product/Nodejs/NodejsTools.vsct

```diff
@@ -225,32 +247,42 @@
         </Strings>
       </Button>
 
-      <Button guid="guidNodeToolsWorkspaceCmdSet" id="cmdidWorkSpaceNpmInstallMissing" priority="0x010" type="Button">
-        <Parent guid="guidNodeToolsWorkspaceCmdSet" id="WorkspaceGroup"/>
+      <Button guid="guidNodeToolsWseContextMenuCmdSet" id="cmdidWorkSpaceNpmInstallMissing" priority="0x010" type="Button">
         <CommandFlag>DefaultInvisible</CommandFlag>
         <CommandFlag>DynamicVisibility</CommandFlag>
         <Strings>
           <ButtonText>Install Missing npm Packages...</ButtonText>
         </Strings>
       </Button>
 
-      <Button guid="guidNodeToolsWorkspaceCmdSet" id="cmdidWorkSpaceNpmInstallNew" priority="0x011" type="Button">
-        <Parent guid="guidNodeToolsWorkspaceCmdSet" id="WorkspaceGroup"/>
+      <Button guid="guidNodeToolsWseContextMenuCmdSet" id="cmdidWorkSpaceNpmInstallNew" priority="0x011" type="Button">
         <CommandFlag>DefaultInvisible</CommandFlag>
         <CommandFlag>DynamicVisibility</CommandFlag>
         <Strings>
           <ButtonText>Install New npm Packages...</ButtonText>
         </Strings>
       </Button>
 
-      <Button guid="guidNodeToolsWorkspaceCmdSet" id="cmdidWorkSpaceNpmUpdate" priority="0x012" type="Button">
-        <Parent guid="guidNodeToolsWorkspaceCmdSet" id="WorkspaceGroup"/>
+      <Button guid="guidNodeToolsWseContextMenuCmdSet" id="cmdidWorkSpaceNpmUpdate" priority="0x012" type="Button">
         <CommandFlag>DefaultInvisible</CommandFlag>
         <CommandFlag>DynamicVisibility</CommandFlag>
         <Strings>
           <ButtonText>Update npm Packages...</ButtonText>
         </Strings>
       </Button>
+      
+      <!-- workspace -->
+      <Button guid="guidNodeToolsWseContextMenuCmdSet" id="cmdidWorkSpaceNpmDynamicScript" priority="0x0130" type="Button">
+        <CommandFlag>DynamicItemStart</CommandFlag>
+        <CommandFlag>NoShowOnMenuController</CommandFlag>
+        <CommandFlag>DefaultInvisible</CommandFlag>
+        <CommandFlag>DynamicVisibility</CommandFlag>
+        <Strings>
+          <ButtonText>Start Script</ButtonText>
+          <CommandName>StartScript</CommandName>
```

Comment:
What is this "StartScript" about?

Reply:
It's a placeholder string, which'll be replaced in the Query calls with the actual name of the scripts.